### PR TITLE
Java exmaples d web search

### PR DESCRIPTION
**Pull request by Zevleg**
MealPlan works but is returning java.lang.IllegalStateException: FAILED ASSERTION: unexpected token: ASSIGN error

Markets returning null raisingStockSymbols and decreasingStockSymbols

BreakingNews is returning java.lang.StackOverflowError

### CODE REVIEW

File: examples/java/src/main/java/com/xebia/functional/xef/java/auto/BreakingNews.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    static LocalDateTime now = LocalDateTime.now();
+
+
+    private static CompletableFuture<Void> writeParagraph(AIScope scope) {
+        var currentDate = dtf.format(now);
+
+        scope.prompt("write a paragraph of about 300 words about: " + currentDate + " Covid News", BreakingNews.class)
+              .thenAccept(breakingNews -> System.out.println(currentDate + " Covid news summary:\n" + breakingNews.summary));
+
+        return writeParagraph(scope);
```
Comment (adam47deg) [inline examples/java/src/main/java/com/xebia/functional/xef/java/auto/BreakingNews.java:22]: writeParagraph is calling itself, so it recurses and runs out of stack space

File: examples/java/src/main/java/com/xebia/functional/xef/java/auto/DivergentTasks.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package com.xebia.functional.xef.java.auto;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+public class DivergentTasks {
+
+    public Long numberOfMedicalNeedlesInWorld;
+
+    private static CompletableFuture<Void> numberOfMedicalNedles(AIScope scope) {
```
Comment (adam47deg) [inline examples/java/src/main/java/com/xebia/functional/xef/java/auto/DivergentTasks.java:10]: rename to "numberOfMedicalNeedles"

File: examples/java/src/main/java/com/xebia/functional/xef/java/auto/Markets.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    @Override
+    public String toString() {
+        return "Markets{" +
+              "news='" + news + '\'' +
+              ", raisingStockSymbols=" + raisingStockSymbols +
+              ", decreasingStockSymbols=" + decreasingStockSymbols +
+              '}';
+    }
+
+    private static CompletableFuture<Void> sumaryOfStockMarket(AIScope scope) {
```
Comment (adam47deg) [inline examples/java/src/main/java/com/xebia/functional/xef/java/auto/Markets.java:23]: rename to "summaryOfStockMarket"

**Review by adam47deg — CHANGES_REQUESTED** [review]
BreakingNews is calling itself in writeParagraph, please double check

File: examples/java/src/main/java/com/xebia/functional/xef/java/auto/BreakingNews.java (hunk 1 above)
Comment (Zevleg) [reply examples/java/src/main/java/com/xebia/functional/xef/java/auto/BreakingNews.java]: Done, but now is returning java.lang.OutOfMemoryError: Java heap space in Java example and Kotlin example

File: examples/java/src/main/java/com/xebia/functional/xef/java/auto/DivergentTasks.java (hunk 2 above)
Comment (Zevleg) [reply examples/java/src/main/java/com/xebia/functional/xef/java/auto/DivergentTasks.java]: Done!

File: examples/java/src/main/java/com/xebia/functional/xef/java/auto/BreakingNews.java (hunk 1 above)
Comment (adam47deg) [reply examples/java/src/main/java/com/xebia/functional/xef/java/auto/BreakingNews.java]: I am not seeing the error anymore

**Review by adam47deg — APPROVED** [review]

**Pull request merged**